Java/Ch 05. Bit Manipulation/Q5_06_Conversion/Question.java

```diff
@@ -5,27 +5,32 @@
 public class Question {
 	public static int bitSwapRequired(int a, int b) {
 		int count = 0;
-		for (int c = a ^ b; c != 0; c = c >> 1) { 
+		for (int c = a ^ b; c != 0; c = c >> 1) {
 			count += c & 1;
 		}
 		return count;
 	}
-	
-	public static int bitSwapRequired2(int a, int b){
+
+	public static int bitSwapRequired2(int a, int b) {
 		int count = 0;
-		for (int c = a ^ b; c != 0; c = c & (c-1)) {
+		for (int c = a ^ b; c != 0; c = c & (c - 1)) {
 			count++;
 		}
 		return count;
 	}
-	
+
+	public static int bitSwapRequired3(int a, int b) {
+		return Integer.bitCount(a ^ b);

```

Comment:
Code for `bitCount`:

```java
        i = i - ((i >>> 1) & 0x55555555);
        i = (i & 0x33333333) + ((i >>> 2) & 0x33333333);
        i = (i + (i >>> 4)) & 0x0f0f0f0f;
        i = i + (i >>> 8);
        i = i + (i >>> 16);
        return i & 0x3f;
```

O(1) for 32-bit integer. And can be generalized for N-bits as well.